### ISSUE TITLE
TransferManager race condition in DoSinglePartDownload

### ISSUE THREAD

**Issue by lynchpancho** [aws/aws-sdk-cpp#1461]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x ] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) and [API reference](http://sdk.amazonaws.com/cpp/api/LATEST/index.html)
- [x ] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-cpp/issues) and didn't find any solution
  
**Describe the bug**
Application code is downloading a small file using transfer manager. Application needs to open the file and verify contents but the file is open in the TransferManager and application code intermittently fails.

**SDK version number**
1.8.46

**Platform/OS/Hardware/Device**
Windows

**To Reproduce (observed behavior)**
```
            auto createStreamFn = [=]() { return  Aws::New<Aws::FStream>("Download", file.c_str(), std::ios_base::out | std::ios_base::in | std::ios_base::binary | std::ios_base::trunc); };
            auto handle = m_TransferManager->DownloadFile(...);
            handle->WaitUntilFinished();
            // open file here
```
TransferManager::DoSinglePartDownload eventually calls handle->UpdateStatus(Transfer::COMPLETED) upon success which releases the application code that is blocked on WaitUntilFinished() but the getObjectOutcome doesn't always go out of scope to close the stream so the application code is failing to open the file because it is still open. 

**Expected behavior**
Stream should be cleaned up before callback releasing client code waiting on WaitUntiFinished(). I believe this is happening in the Multipart case.

**Logs/output**

To enable logging, set the following system properties:

*REMEMBER TO SANITIZE YOUR PERSONAL INFO*

```
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
Aws::InitAPI(options)
```

**Additional context**
Thank you!
